src/PositionManager.sol

```diff
@@ -204,6 +207,14 @@ contract PositionManager is
         }
     }
 
+    /// @dev integrators may elect to forfeit positive deltas with clear
+    /// provide a safety check that amount-to-clear is less than a user-provided maximum
+    function _clear(Currency currency, uint256 amountMax) internal {
+        int256 currencyDelta = poolManager.currencyDelta(address(this), currency);
+        if (uint256(currencyDelta) > amountMax) revert ClearExceedsMaxAmount(currency, currencyDelta, amountMax);
```

Comment:
more a stylistic thing - but I wonder if the behaviour should be that

```
if (uint256(currencyDelta) > amountMax) _take(currency, amount)
```

i feel like the natural behaviour is "if its less than X its cheaper for me to leave it, but if its more than X I want it" ?

Reply:
from offline: we'll introduce CLEAR_OR_TAKE as a command. we want to try to map commands ~1:1 with pool manager